code/game/machinery/announcement_system.dm

```diff
@@ -153,6 +153,9 @@ GLOBAL_LIST_EMPTY(announcement_systems)
 	add_fingerprint(usr)
 
 /obj/machinery/announcement_system/attack_robot(mob/living/silicon/user)
+	. = ..()
+	if(.)
+		return
```

Comment:
This is a pattern you use a lot, and it seems like it would lead to some fucky behavior. Enforcing the parent call passthrough makes me a bit uncomfortable

Reply:
My reasoning was that the proc call chain needs to respect a signal handler returning `COMPONENT_CANCEL_ATTACK_CHAIN`

Reply:
I mean the parent call in the first place. Since the base of `attack_robot` calls `attack_ai`, and the code you're modfiying may not actually want that. Seems A: like a limitation on what attack_robot can be used for, and B: like it would break some of these procs and what they intend to do

Reply:
So far, only the supermatter uses COMSIG_ATOM_ATTACK_ROBOT for anything, and it worked as intended in that case. It really depends on how future uses of COMSIG_ATOM_ATTACK_ROBOT pan out.

Reply:
Sorry I honestly kinda forgot about this.
I mean that the parent call of `attack_robot` does stuff outside of sending a signal. If memory serves it calls attack_ai, which in this instance means doubling up on ai_attacks.

I'm concerned that this will lead to changed behavior, since you're treating it as if it just sends a parent call.